File: CardNews/example2/screenshot.js
const puppeteer = require('puppeteer');
const path = require('path');

(async () => {
  const browser = await puppeteer.launch({
    headless: 'new'
  });

  const page = await browser.newPage();

  // Set viewport to match the card size
  await page.setViewport({
    width: 720,
    height: 720,
    deviceScaleFactor: 2
  });

  // Navigate to the HTML file
  const htmlPath = path.join(__dirname, 'card_v1.html');
  await page.goto(`file://${htmlPath}`, {
    waitUntil: 'networkidle0'
  });

  // Wait a bit for fonts to load
  await page.waitForTimeout(1000);

  // Take screenshot
  await page.screenshot({
    path: path.join(__dirname, 'outputs', 'output_v1.png'),
    fullPage: false
  });

  console.log('Screenshot saved to outputs/output_v1.png');

  await browser.close();
})();

File: CardNews/example2/screenshot.py
#!/usr/bin/env python3
from playwright.sync_api import sync_playwright
import os

def take_screenshot():
    script_dir = os.path.dirname(os.path.abspath(__file__))
    html_path = os.path.join(script_dir, 'card_v1.html')
    output_path = os.path.join(script_dir, 'outputs', 'output_v1.png')

    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page(viewport={'width': 720, 'height': 720})

        page.goto(f'file://{html_path}')
        page.wait_for_timeout(1000)  # Wait for fonts to load

        page.screenshot(path=output_path, full_page=False)

        print(f'Screenshot saved to {output_path}')
        browser.close()

if __name__ == '__main__':
    take_screenshot()
